sycl/plugins/cuda/pi_cuda.hpp

```diff
@@ -639,16 +641,11 @@ struct _pi_kernel {
       std::memcpy(implicitOffsetArgs_, implicitOffset, size);
     }
 
-    void clear_local_size() {
-      std::fill(std::begin(offsetPerIndex_), std::end(offsetPerIndex_), 0);
-    }
+    void clear_local_size() { offsetSum = 0; }
```

Comment:
mark this inline noexcept

```suggestion
    inline void clear_local_size() noexcept { offsetSum = 0; }
```